services/scabbard/libscabbard/src/service/factory/factory_database.rs

```diff
@@ -424,33 +422,35 @@ impl ServiceFactory for ScabbardFactory {
             )
         };
 
-        let (receipt_store, commit_hash_store): (ScabbardReceiptStore, Box<dyn CommitHashStore>) =
-            match &self.store_factory_config {
-                #[cfg(feature = "postgres")]
-                ScabbardFactoryStorageConfig::Postgres { pool } => (
-                    Arc::new(RwLock::new(DieselReceiptStore::new(
-                        pool.clone(),
-                        Some(format!("{}::{}", circuit_id, service_id)),
-                    ))),
-                    Box::new(DieselCommitHashStore::new(
-                        pool.clone(),
-                        circuit_id,
-                        &service_id,
-                    )),
-                ),
-                #[cfg(feature = "sqlite")]
-                ScabbardFactoryStorageConfig::Sqlite { pool } => (
-                    Arc::new(RwLock::new(DieselReceiptStore::new(
-                        pool.clone(),
-                        Some(format!("{}::{}", circuit_id, service_id)),
-                    ))),
-                    Box::new(DieselCommitHashStore::new(
-                        pool.clone(),
-                        circuit_id,
-                        &service_id,
-                    )),
-                ),
-            };
+        let (receipt_store, commit_hash_store): (
+            Arc<RwLock<dyn ReceiptStore>>,
```

Comment:
Since you’re removing the type, might as well remove the arc-mutex combo here as it’s not necessary here (in another commit)

Reply:
Added a commit to remove the RwLock